docs/source/autoscaling.rst

```diff
@@ -12,21 +12,45 @@ Enabling autoscaling
 
-Autoscaling parameters are stored in an ``autoscaling`` attribute of your instances as a dictionary. Within the ``autoscaling`` attribute, specifying a ``method`` will allow you to specify the logic that determines when to scale up and scale down your instance. If an autoscaling method isn't provided, the ``"default"`` autoscaling method will be used.
+Autoscaling parameters are stored in an ``autoscaling`` attribute of your instances as a dictionary. Within the ``autoscaling`` attribute, setting an ``ingester`` will allow you to specify a method that determines the utilization of your service. If an ingester isn't provided, the ``"mesos_cpu_ram"`` ingester will be used. Within the ``autoscaling`` attribute, setting a ``decider`` will allow you to specify the logic that determines when to autoscale your service. If a decider isn't provided, the ``"pid"`` decider will be used. Deciders and ingesters have their own optional keyword arguments that may be placed into the ``autoscaling`` dictionary as well.
 
-Autoscaling methods
--------------------
+Autoscaling components
+----------------------
 
-The current autoscaling methods are:
+The currently available ingesters are:
 
-:default:
-  The default autoscaling method if none is provided. Tries to use cpu and ram usage to predict when to autoscale. (NOT YET IMPLEMENTED)
+:mesos_cpu_ram:
+  The default autoscaling method if none is provided. Tries to use cpu and ram usage to predict when to autoscale.
+:http:
+  Makes a request on a HTTP endpoint on your service. Expects a JSON-formatted dictionary with a 'utilization' field containing a number between 0 and 1.
+
+  Autoscaling parameters:
+
+  :endpoint: the path to perform the HTTP request on (the requested URL will be \http://$HOST:$PORT/endpoint). Defaults to 'status'.
+
+The currently available deciders are:
+
+:pid:
+  Uses a PID controller to determine when to autoscale a service.
+
+  Autoscaling parameters:
+
+  :setpoint: the target utilization the controller aims for. Defaults to 0.8 (80%).
+  :delay: the number of seconds the decider must wait before fetching new data. Defaults to 600.
+:threshold: 
+  Autoscales when a service's utilization exceeds beyond a certain threshold.
+  
+  Autoscaling parameters:
+
+  :setpoint: the target utilization the controller aims for. Defaults to 0.8 (80%).
+  :threshold: the amount by which the setpoint must be exceeded in either direction before autoscaling is triggered. Defaults to 0.1 (10%).
+  :delay: the number of seconds the decider must wait before fetching new data. Defaults to 600.
```

Comment:
Could this delay be per-action? I'm thinking that you might want different delays for removing instances to adding instances - it might take significantly longer to bring a new instance up than to shut it down.